packages/dev/core/src/Materials/greasedLineSimpleMaterial.ts

```diff
@@ -0,0 +1,511 @@
+import type { Scene } from "../scene";
+import { Engine } from "../Engines/engine";
+import { RawTexture } from "./Textures/rawTexture";
+import type { GreasedLineMaterialOptions, IGreasedLineMaterial } from "./greasedLinePluginMaterial";
+import { GreasedLineMeshColorMode, GreasedLineMeshColorDistributionType } from "./greasedLinePluginMaterial";
+
+import { ShaderMaterial } from "./shaderMaterial";
+import type { Nullable } from "../types";
+import { Color3 } from "../Maths/math.color";
+import { TmpVectors, Vector2 } from "../Maths/math.vector";
+import type { BaseTexture } from "./Textures/baseTexture";
+
+import "../Shaders/greasedLine.fragment";
+import "../Shaders/greasedLine.vertex";
+
+/**
+ * GreasedLineSimpleMaterial
+ */
+export class GreasedLineSimpleMaterial extends ShaderMaterial implements IGreasedLineMaterial {
+    /**
+     * Default line color for newly created lines
+     */
+    public static DEFAULT_COLOR = Color3.White();
+    /**
+     * Default line width when sizeAttenuation is true
+     */
+    public static DEFAULT_WIDTH_ATTENUATED = 1;
+    /**
+     * Defaule line width
+     */
+    public static DEFAULT_WIDTH = 0.1;
+
+    private static _EmptyColorsTexture: BaseTexture;

```

Comment:
is this texture ever disposed? Does it stay in memory when disposing the engine?

Reply:
Added an engine disposed observer.

Reply:
commented on the added function